README.md

```diff
@@ -3,9 +3,35 @@
 
 # OpenQA Public cloud Helper
 
-PublicCloud-Watcher or `pcw` is a web app which monitors, displays and deletes resources on various Cloud Service Providers (CSPs). It identifies abandoned instances by searching for certain tags and performs cleanup tasks. It also deletes old custom uploaded images. The behavior differs per CSP.
-
-The fastest way to run PublicCloud-Watcher is via the provided containers, as described in the [Running a container](#running-a-container) section.
+PublicCloud-Watcher (PCW) is a web app which monitors, displays and deletes resources on various Cloud Service Providers (CSPs).
+PCW has two main flows :
+
+1. Update run ( implemented in [ocw/lib/db.py](ocw/lib/db.py) )
+ Concentrates on deleting VMs (in case of Azure Resource Groups) . Each update run scans accounts defined in configuration file and
+and persists all results of each scan into sqlite database. If it is new entity than it obligatory gets TTL. TTL may be taken from tag `openqa_ttl` if
+entity is tagged with such tag if not PCW will check `pcw.ini` for `updaterun/default_ttl` setting if setting is not defined than PCW will use hard-coded value from [webui/settings.py](webui/settings.py). Database has a web UI where you can manually trigger certain entity deletion.
+
+ After persisting results into db PCW deciding which entities needs to be deleted. There are two ways to survive for entity :
+   a. Having tag `pcw_ignore` ( with any value)
+   b. Age of entity is lower than TTL defined. Age is calculated as delta of last_seen and first_seen
+
+  For entities that survive cleanup PCW will sent notification email to the list defined in config.
+
+2. Cleanup ( implemented in [ocw/lib/cleanup.py](ocw/lib/cleanup.py) )
+ Concentrates on everything except VM deletion . This vary a lot per CSP so let's clarify that on per provider level.
```

Comment:
"Concentrates on everything except VM deletion" is only useful for people who know already the internals and know what "everything except VM deletion" means. I have no idea what that means, but based on the text below I assume it means the deletion of dangling resources.

```suggestion
 Concentrates on deleting dangling resources from the previously scannerd virtual machines, e.g. disk images. The actual behaviour varies per CSP:
```

Reply:
ehm but to know this you need to read few lines above . I assume I may rely on a fact that person read whole README and should not phrase text expecting every sentence to be self-containing

Reply:
Not in general, but the sentence at the beginning of a paragraph should contain the most important message of that paragraph. Those sentences should describe what is going on and why the paragraph is important. The first sentence should be indeed self-containing.

This is arguably the most important/condensed piece of information and if the reader is already starved for contextual information here, then the read flow is disturbed and most readers annoyed.

Reply:
it is not paragraph but point of the list which is very important difference in context of our conversation . So basically you have **point of the list** number 1 saying "**Update run ( implemented in [ocw/lib/db.py](ocw/lib/db.py) )** Concentrates on deleting VMs (in case of Azure Resource Groups)."  and point of the list number 2 saying "**Cleanup ( implemented in [ocw/lib/cleanup.py](ocw/lib/cleanup.py) )** Concentrates on everything except VM deletion.". To me the list is self-containing . I am insisting on such wording because for me it is part of whole architectural desing of PCW which has two main threads and one thread concentrates on deletion of VMs and another on **everything else** so whenever we would decide to add another entity to monitor it will gets into cleanup htread and not into update run ( or whatever new name this thread will have )

Reply:
To come back to my original comment here: IMHO "Concentrates on everything except VM deletion" is too unspecific and doesn't give the reader any new information. It should be more specific on what the "everything else is", as suggested in my original comment.

I leave it however up to you to accept it or leave it as it is.